autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/sync/CredentialsSyncStore.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.autofill.sync
+
+import android.content.Context
+import android.content.SharedPreferences
+import androidx.core.content.edit
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+interface CredentialsSyncStore {
+    var serverModifiedSince: String
+    var startTimeStamp: String
+    var clientModifiedSince: String
+}
+
+@ContributesBinding(AppScope::class)
+@SingleInstanceIn(AppScope::class)
+class RealCredentialsSyncStore @Inject constructor(private val context: Context) : CredentialsSyncStore {
+    override var serverModifiedSince: String
+        get() = preferences.getString(KEY_MODIFIED_SINCE, "0") ?: "0"
+        set(value) = preferences.edit(true) { putString(KEY_MODIFIED_SINCE, value) }
+    override var startTimeStamp: String
+        get() = preferences.getString(KEY_START_TIMESTAMP, "0") ?: "0"
+        set(value) = preferences.edit(true) { putString(KEY_START_TIMESTAMP, value) }
+    override var clientModifiedSince: String
+        get() = preferences.getString(KEY_END_TIMESTAMP, "0") ?: "0"
+        set(value) = preferences.edit(true) { putString(KEY_END_TIMESTAMP, value) }
+
+    private val preferences: SharedPreferences
+        get() = context.getSharedPreferences(FILENAME, Context.MODE_PRIVATE)
+
+    companion object {
+        const val FILENAME = "com.duckduckgo.credentials.sync.store"
```

Comment:
not a big deal, but should we match the file's name package to the filename here?